### PR TITLE
Enable Samsung backend for ndk 25 build

### PR DESCRIPTION
**Pull request by AhmedTElthakeb**

### CODE REVIEW

**Comment by anhappdev** [comment]
There are still some errors:

From App Build / build-android-apk:
```
Validate checksum of Samsung lib files
flutter/tool/validate-checksum.sh \
	-d mobile_back_samsung/samsung/lib \
FAIL: internal/libc++_shared.so (File not found)
	-f mobile_back_samsung/samsung/lib/checksums.txt
make: *** [flutter/android/android.mk:[31](https://github.com/mlcommons/mobile_app_open/actions/runs/6784137159/job/18440174294?pr=812#step:11:32): flutter/android/libs/checksum] Error 1
```


From Code Analysis / SonarScanner:
```
ERROR: /image-workdir/project/mobile_back_pixel/cpp/backend_tflite/BUILD:86:18: Linking mobile_back_pixel/cpp/backend_tflite/libtflitepixelbackend.so failed: (Exit 1): clang failed: error executing command (from target //mobile_back_pixel/cpp/backend_tflite:libtflitepixelbackend.so) 
  (cd /image-workdir/.cache/bazel/_bazel_root/f96fd4f8f5bca05be4940a7079fbe1d9/execroot/mlperf_app && \
  exec env - \
    PATH=/image-workdir/.cache/bazelisk/downloads/bazelbuild/bazel-6.3.2-linux-x86_64/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/android/cmdline-tools/tools/bin:/image-workdir/flutter/bin:/image-workdir/flutter/bin/cache/dart-sdk/bin:/image-workdir/.cache/.pub-cache/bin:/image-workdir/.sonar/sonar-scanner-5.0.1.3006-linux/bin:/image-workdir/.sonar/build-wrapper-linux-x86 \
    *** \
  external/androidndk/toolchains/llvm/prebuilt/linux-x86_64/bin/clang @bazel-out/arm64-v8a-opt/bin/mobile_back_pixel/cpp/backend_tflite/libtflitepixelbackend.so-2.params)
# Configuration: f70a72596caa7291b35dc91e1273c7f9c14dcfa7dd8606bbd44471844ef2ad94
# Execution platform: @local_execution_config_platform//:platform
ld.lld: error: unknown -z value: separate-code
ld.lld: error: cannot open crtbegin_so.o: No such file or directory
ld.lld: error: unable to find library -lEGL
ld.lld: error: unable to find library -lGLESv2
ld.lld: error: unable to find library -lnativewindow
ld.lld: error: unable to find library -llog
ld.lld: error: cannot open crtend_so.o: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
INFO: Elapsed time: 701.875s, Critical Path: 30.74s
INFO: 2096 processes: 281 internal, 1815 local.
FAILED: Build did NOT complete successfully
make[1]: *** [flutter/android/android.mk:40: flutter/android/libs/build] Error 1
```

**Comment by anhappdev** [comment]
@AhmedTElthakeb The master branch was merged into `ndk_25_quick_hack`, and now there is a conflicting file:
`mobile_back_samsung/samsung/lib/checksums.txt`
Please resolve the conflict. If you have updated *.so files, please also provide a link to them.

**Comment by AhmedTElthakeb** [comment]
> @AhmedTElthakeb The master branch was merged into `ndk_25_quick_hack`, and now there is a conflicting file: `mobile_back_samsung/samsung/lib/checksums.txt` Please resolve the conflict. If you have updated *.so files, please also provide a link to them.

I updated the checksum, should be okay to merge with Master now.

**Comment by anhappdev** [comment]
@AhmedTElthakeb It looks like you didn't pull the latest `ndk_25_quick_hack`. The file `checksums.txt` is still in conflict.

**Comment by AhmedTElthakeb** [comment]
> @AhmedTElthakeb The master branch was merged into `ndk_25_quick_hack`, and now there is a conflicting file: `mobile_back_samsung/samsung/lib/checksums.txt` Please resolve the conflict. If you have updated *.so files, please also provide a link to them.

Actually some .so libs got updated, so I updated the [README.md](https://github.com/mlcommons/mobile_app_open/blob/samsung_backend_for_ndk25/mobile_back_samsung/README.md) with a pointer to the new [libs](https://github.com/mlcommons/mobile_back_samsung/tree/submission_v3.1_samsung_backend_ndk25/samsung_libs).

File: mobile_back_samsung/samsung/lib/checksums.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
 e2f78216c1a3e0d925e17e4d6e7befbc1a2a853867864c6f30e553bf971f2a42  internal/libc++.so
 34ce517c63720a6a858d4e0237247a4e6db857cc4a1f6dffe7210916b96fdc3b  internal/libeden_nn_on_system.so
-f8a6c9d5c0e094c8ef84468ee535ff2f4604d9e59064925aa38cfdd200a06392  internal/libenn_extension.so
+f459d2b74a2330b938ea163037c5c5f896a97c1b3a7c1454051246bdab766e5c  internal/libenn_extension.so
```
Comment (anhappdev) [inline mobile_back_samsung/samsung/lib/checksums.txt:3]: The checksum of `internal/libenn_extension.so` in 
https://github.com/mlcommons/mobile_back_samsung/tree/submission_v3.1_samsung_backend_ndk25/samsung_libs) does not match. Please check again.

**Review by anhappdev — CHANGES_REQUESTED** [review]

**Pull request merged**